projects.json

```diff
@@ -167,7 +167,7 @@
     "compatibility": [
       {
         "version": "4.0",
```

Comment:
@igor-makarov Thanks for setting this up! Would you mind also bumping the version to 4.2? I've verified that the baseline passes by running project_precommit_check, and also that it builds with the compiler built from swift:master branch.

```
# project_precommit_check using swiftc in Xcode 10 Beta 6
./project_precommit_check BeaconKit --earliest-compatible-swift-version 4.2
** CHECK **
--- Validating BeaconKit Swift version 4.2 compatibility ---
--- Project configured to be compatible with Swift 4.2 ---
--- Checking BeaconKit platform compatibility with Darwin ---
--- Platform compatibility check succeeded ---
--- Locating swiftc executable ---
$ xcrun -f swiftc
--- Checking installed Swift version ---
$ /Applications/Xcodes/Xcode_10_Beta_6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc --version
--- Version check succeeded ---
--- Executing build actions ---
$ ./runner.py --swift-branch swift-4.2-branch --swiftc /Applications/Xcodes/Xcode_10_Beta_6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc --projects projects.json --include-repos 'path == "BeaconKit"' --include-versions 'version == "4.2"' --include-actions 'action.startswith("Build")'
PASS: BeaconKit, 4.2, 6e6b0f, Swift Package
========================================
Action Summary:
     Passed: 1
     Failed: 0
    XFailed: 0
    UPassed: 0
      Total: 1
========================================
Repository Summary:
      Total: 1
========================================
Result: PASS
========================================
--- BeaconKit checked successfully against Swift 4.2 ---

# Test using swiftc built from swift:master
./runner.py --swiftc build/compat_macos/install/toolchain/usr/bin/swiftc --projects projects.json --include-actions 'action.startswith("Build")' --swift-branch master --include-repos 'path == "BeaconKit"'
PASS: BeaconKit, 4.2, 6e6b0f, Swift Package
========================================
Action Summary:
     Passed: 1
     Failed: 0
    XFailed: 0
    UPassed: 0
      Total: 1
========================================
Repository Summary:
      Total: 1
========================================
Result: PASS
========================================
```